src/Aspire.Hosting/Codespaces/CodespacesUrlRewriter.cs

```diff
@@ -1,76 +1,32 @@
+internal sealed class CodespacesUrlRewriter(IOptions<CodespacesOptions> options)
 {
-    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    public string RewriteUrl(string url)
     {
+        ArgumentNullException.ThrowIfNullOrWhiteSpace(url);
+
         if (!options.Value.IsCodespace)
         {
-            logger.LogTrace("Not running in Codespaces, skipping URL rewriting.");
-            return;
+            throw new InvalidOperationException("Cannot translate URL because not currently running in a GitHub Codespace.");
         }
 
-        do
-        {
-            try
-            {
-                var resourceEvents = resourceNotificationService.WatchAsync(stoppingToken);
-
-                await foreach (var resourceEvent in resourceEvents.ConfigureAwait(false))
-                {
-                    Dictionary<UrlSnapshot, UrlSnapshot>? remappedUrls = null;
-
-                    foreach (var originalUrlSnapshot in resourceEvent.Snapshot.Urls)
-                    {
-                        var uri = new Uri(originalUrlSnapshot.Url);
-
-                        if (!originalUrlSnapshot.IsInternal && (uri.Scheme == "http" || uri.Scheme == "https") && uri.Host == "localhost")
-                        {
-                            remappedUrls ??= new();
-
-                            var newUrlSnapshot = originalUrlSnapshot with
-                            {
-                                // The format of GitHub Codespaces URLs comprises the codespace
-                                // name (from the CODESPACE_NAME environment variable, the port,
-                                // and the port forwarding domain (via GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN
-                                // which is typically ".app.github.dev". The VSCode instance is typically
-                                // hosted at codespacename.github.dev whereas the forwarded ports
-                                // would be at codespacename-port.app.github.dev.
-                                Url = $"{uri.Scheme}://{options.Value.CodespaceName}-{uri.Port}.{options.Value.PortForwardingDomain}{uri.AbsolutePath}"
-                            };
-
-                            remappedUrls.Add(originalUrlSnapshot, newUrlSnapshot);
-                        }
-                    }
+        return RewriteUrl(new Uri(url, UriKind.Absolute));
+    }
 
-                    if (remappedUrls is not null)
-                    {
-                        var transformedUrls = from originalUrl in resourceEvent.Snapshot.Urls
-                                              select remappedUrls.TryGetValue(originalUrl, out var remappedUrl) ? remappedUrl : originalUrl;
+    public string RewriteUrl(Uri uri)
+    {
+        if (!options.Value.IsCodespace)
+        {
+            throw new InvalidOperationException("Cannot translate URL because not currently running in a GitHub Codespace.");
```

Comment:
Why is this throwing? We don't want to stop execution, just log and skip

Reply:
This method shouldn't get called unless we are running in a codespace.

Reply:
Or it could check and return the input url